lib/ansible/parsing/quoting.py

```diff
@@ -25,7 +27,20 @@ def is_quoted(data):
 
 
 def unquote(data):
-    ''' removes first and last quotes from a string, if the string starts and ends with the same quotes '''
+    """Removes first and last quotes from a string, if the string starts and
+    ends with the same quotes.
+
+    :parama data: The string to modify
+    :type data: str
+
+    :return: The string without quotes
+    :rtype: str
+    """
+    if not isinstance(data, six.string_types) or not isinstance(
```

Comment:
To attempt clarifying what I mean here, I believe this `isinstance` check should be moved to the `is_quoted` function.

Reply:
Ok sorry, I'm going to make this modification also.

Reply:
Thanks, it's fixed.